src/component/Signalment.js

```diff
@@ -1,59 +1,78 @@
 import React from 'react';
+import {connect} from 'react-redux';
 import style from 'antd/dist/antd.css';
 import {Form, Col,Select, notification, Input, Switch, Button,Icon } from 'antd';
 import {BrowserRouter,Route,Link} from 'react-router-dom';
+import {store} from '../store/store';
+import {usersignalmentDetails} from '../action/action'
 
+class SignalmentForm extends React.Component {
 
-class SingnalmentForm extends React.Component {
-
-  onChange = (checked)=> {
-    console.log(`switch to ${checked}`);
+  constructor(props) {
+    super(props);
+    this.state = {
+      Distance:props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].Distance,
+      Pension:props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].Pension,
+      Qualification:props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].Qualification,
+      Working_Shift:props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].Working_Shift,
+      facilities:props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].facilities,
+      driving_license:props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].driving_license,
+      own_car:props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].own_car,
+      Location: props.signalmentDetails[props.signalmentDetails.length-1] && props.signalmentDetails[props.signalmentDetails.length-1].Location
+    };
+  }
+  onChange = (defauUnChecked)=> {
+    this.state.driving_license = defauUnChecked;
+  }
+  onChangecar = (check)=> {
+    this.state.own_car =  check;
   }
 
   handleSubmit = (e) => {
     e.preventDefault();
     this.props.form.validateFields((err, values) => {
-
       if (!err) {
-        console.log('Received values of form: ', values);
+         store.dispatch(usersignalmentDetails(values,this.state.driving_license,this.state.own_car));
```

Comment:
change this is using  disptachtoprops which i told u

Reply:
ok sir